packages/interactive-messages/src/adapter.ts

```diff
@@ -454,10 +454,12 @@ export class SlackMessageAdapter {
       // if the callback ID constraint is specified, only continue if it matches
       if (!isFalsy(constraints.callbackId)) {
         // The callback ID is located at a different path in the payload for view submission and view closed
-        const callbackId = (
+        // than for actions
+        const callbackId = ((
           constraints.handlerType === StoredConstraintsType.ViewSubmission ||
           constraints.handlerType === StoredConstraintsType.ViewClosed
-        ) ? payload.view.callback_id : payload.callback_id;
+        ) && payload.view && payload.view.callback_id) ||
+        (constraints.handlerType === StoredConstraintsType.Action && payload.callback_id);
```

Comment:
the multiple levels of embedded short-circuiting logic operators used to express conditional logic makes things quite hard to read. can we expand this into `if`/`else` conditional branches?

Reply:
sure we can!

Reply:
@aoberoi I reverted to the ternary assignment we had earlier, but added a check for `payload.view`